index.js

```diff
@@ -26,155 +38,189 @@ function toggleSound(){
         newCard.appendChild(img1);
         newCard.appendChild(img2);
 
-        document.querySelector('.memory-game').appendChild(newCard); 
+        document.querySelector(".memory-game").appendChild(newCard);
+      }
+      //generate the card array from the image sources
+      cardArray = cardImageSrcs.slice(0, noOfCards / 2);
+      cardArray.push(...cardArray);
+      document.querySelectorAll(".memory-card").forEach((card, i) => {
+        card.addEventListener("click", (e) => flipCard(e, i)); // Passing index of card when calling flipCard
+      });
+      popup_box.classList.remove("popupBox");
+      setTimer();
+    });
+  });
+
+  //increasing the counter
+  let counter = 0;
+  function counterIncrease() {
+    counter++;
+    console.log();
+    timer.innerHTML = "<b>" + counter + "</b>";
+  }
+
+  let interval;
+  function setTimer() {
+    interval = setInterval(counterIncrease, 1000);
+  }
+
+  // open popup model let you choose
+  document.getElementById("start-btn").addEventListener("click", function () {
+    removeChild();
+    counter = 0;
+    clearInterval(interval);
+    timer.innerHTML = "<b>" + 0 + "</b>";
+    popup_box.classList.add("popupBox");
+  });

```

Comment:
Here **removeChild** function is for deleting previously stored cards because every time user chooses a new level, the cards merge with previous cards

Reply:
Resolved the issue! Please consider the last commit